### DISCUSSION TITLE
Event bus

### DISCUSSION THREAD

**Discussion by murilolivorato** [quasarframework/quasar#15480] [General - Components / Directives / etc] [answered]
hello , I am trying to make an event bus like this example -

https://quasar.dev/quasar-utils/event-bus-util 

I can see that has two ways to use that  , one way is making a boot and inject that class .
other way is making importing the class - { EventBus } from 'quasar' .
and using it .
well  , both ways are not working .
I am working with composition api , with - <script setup>

so , I did like that . using the boot -

```
import { EventBus } from 'quasar'
import { boot } from 'quasar/wrappers'

export default boot(({ app }) => {
  const bus = new EventBus()

  // for Options API
  app.config.globalProperties.$bus = bus

  // for Composition API
  app.provide('bus', bus)
})

```

one page -

```
// Composition API
import { inject } from 'vue'

const bus = inject('bus') 
bus.emit('myevent')
```

in my component -

```
import { inject } from 'vue'
const bus = inject('bus') 
bus.on('myevent', () => {
  console.log('myevent received!')
})

```
result -
```
can not read property of undefined . ( reading 'emit')
```

OTHER WAY IMPORTING CLASS EVENT BUSS -

in page -

```
import { EventBus } from 'quasar'
const bus = new EventBus()
  bus.emit('myevent')
```
in my component -

```
import { EventBus } from 'quasar'
const bus = new EventBus()
bus.on('myevent', () => {
  console.log('myevent received!')
})
```
It does not shoes any erors , but does not listen the event .

**Comment by dongwa** [comment, marked as answer]
If you are  using  **Composition api**,the most convenient way is create a `ts/js` file,and `new EventBus()` ,export it,then you can import this bus in anywhere.
Not need a boot file,a boot file is more suitable for Options API
## example：
``` js
// bus.js or or anywhere you like
import  { EventBus }  from 'quasar'
const bus = new EventBus()
export bus
```
``` js
// some component or page
import { bus } from 'path-to-your-bus'
bus.on('xxx',()=>{
  console.log('xxx is been emit')
})
```
``` js
// another component or page
import { bus } from 'path-to-your-bus'
bus.emit('xxx')
```
